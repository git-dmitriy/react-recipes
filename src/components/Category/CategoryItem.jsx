import { Link } from 'react-router-dom';

export default function CategoryItem({
  idCategory,
  strCategory,
  strCategoryThumb,
  strCategoryDescription,
}) {
  return (
    <div class='card'>
      <div class='card-image'>
        <img src={strCategoryThumb} alt={strCategory} />
      </div>
      <div class='card-content'>
        <span class='card-title'>{strCategory}</span>
        <p>{strCategoryDescription.slice(0, 60)}...</p>
      </div>
      <div className='card-action'>
        <Link to={`/category/${idCategory}`} className='btn'>
          Watch category
        </Link>
      </div>
    </div>
  );
}
