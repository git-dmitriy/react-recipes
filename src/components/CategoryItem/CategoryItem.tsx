import {Link} from 'react-router-dom';
// import LazyLoad from 'react-lazyload';
import {CategoryItemTypes} from '@/appTypes.ts';

export const CategoryItem: React.FC<CategoryItemTypes> = ({
                                                              strCategory,
                                                              strCategoryThumb,
                                                              strCategoryDescription,
                                                          }) => {
    return (
        <article
            className='rounded-3xl border border-gray-200 dark:border-gray-900 overflow-hidden bg-white dark:bg-orange-100 dark:text-gray-900 sm:mx-0 min-h-170p'>
            <Link
                to={`/category/${strCategory}`}
                className='h-full flex hover:opacity-90 transition-opacity'
            >
                <div className='w-2/5 sm:w-2/4 bg-white relative'>
                    {/*<LazyLoad*/}
                    {/*    height={180}*/}
                    {/*    offset={400}*/}
                    {/*    once*/}
                    {/*    classNamePrefix='h-full bg-gray-200 '*/}
                    {/*>*/}
                    <img
                        className='h-full object-cover'
                        width='320'
                        height='200'
                        src={strCategoryThumb}
                        alt={strCategory}
                    />
                    {/*</LazyLoad>*/}
                </div>
                <div className='w-3/5 sm:w-2/4 flex flex-col items-start mx-4 my-4'>
                    <h1 className='text-xl font-semibold mb-5 sm:mb-0'>{strCategory}</h1>
                    <p className='text-sm sm:text-md'>
                        {strCategoryDescription.length > 50
                            ? strCategoryDescription.slice(0, 120) + '...'
                            : strCategoryDescription}
                    </p>
                </div>
            </Link>
        </article>
    );
};
