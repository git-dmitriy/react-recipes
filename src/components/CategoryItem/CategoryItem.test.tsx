import renderer from 'react-test-renderer';
import {CategoryItem} from './CategoryItem';
import {MemoryRouter} from 'react-router-dom';
import {it, expect} from 'vitest'

const data = {
    idCategory: '13',
    strCategory: 'Breakfast',
    strCategoryDescription:
        'Breakfast is the first meal of a day. The word in English refers to breaking the fasting period of the previous night. There is a strong likelihood for one or more "typical", or "traditional", breakfast menus to exist in most places, but their composition varies widely from place to place, and has varied over time, so that globally a very wide range of preparations and ingredients are now associated with breakfast.',
    strCategoryThumb: 'https://www.themealdb.com/images/category/breakfast.png',
};

it('should render correctly', () => {
    const tree = renderer
        .create(
            <MemoryRouter>
                <CategoryItem {...data} />
            </MemoryRouter>
        )
        .toJSON();

    expect(tree).toMatchSnapshot();
});
