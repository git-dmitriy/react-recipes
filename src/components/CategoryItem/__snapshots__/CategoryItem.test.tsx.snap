// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render correctly 1`] = `
<article
  className="rounded-3xl border border-gray-200 dark:border-gray-900 overflow-hidden bg-white dark:bg-orange-100 dark:text-gray-900 sm:mx-0 min-h-170p"
>
  <a
    className="h-full flex hover:opacity-90 transition-opacity"
    href="/category/Breakfast"
    onClick={[Function]}
  >
    <div
      className="w-2/5 sm:w-2/4 bg-white relative"
    >
      <img
        alt="Breakfast"
        className="h-full object-cover"
        height="200"
        src="https://www.themealdb.com/images/category/breakfast.png"
        width="320"
      />
    </div>
    <div
      className="w-3/5 sm:w-2/4 flex flex-col items-start mx-4 my-4"
    >
      <h1
        className="text-xl font-semibold mb-5 sm:mb-0"
      >
        Breakfast
      </h1>
      <p
        className="text-sm sm:text-md"
      >
        Breakfast is the first meal of a day. The word in English refers to breaking the fasting period of the previous night. T...
      </p>
    </div>
  </a>
</article>
`;
